assets/app/view/game/spreadsheet.rb

```diff
@@ -183,40 +191,51 @@ def render_corporations
         current_round = @game.turn_round_num
 
         sorted_corporations.map.with_index do |corp_array, index|
-          render_corporation(corp_array[1], corp_array[0], current_round, index)
+          render_corporation(*corp_array, current_round, index)
         end
       end
 
       def sorted_corporations
         floated_corporations = @game.round.entities
 
-        result = @game.all_corporations.map do |c|
+        result = @game.all_corporations.reject(&:closed?).select { |c| c.minor? || c.ipoed }
+        result = result.sort.each.with_index.map do |c, order|
           operating_order = (floated_corporations.find_index(c) || -1) + 1
-          [operating_order, c]
+          [c, operating_order, order + 1]
         end
 
-        result.sort_by! do |operating_order, corporation|
-          case @spreadsheet_sort_by
-          when :cash
-            corporation.cash
-          when :id
-            corporation.id
-          when :order
-            operating_order
-          when :par_price
-            corporation.par_price&.price || 0
-          when :share_price
-            corporation.share_price&.price || 0
-          else
-            @game.player_by_id(@spreadsheet_sort_by)&.num_shares_of(corporation)
-          end
+        result = result.map { |c, _, order| [c, order, order] } if result.to_enum.with_object(1).map(&:[]).all?(0)
+
+        result.sort_by! do |corporation, operating_order, order|
+          [case @spreadsheet_sort_by
+           when :cash
+             corporation.cash
+           when :id
+             corporation.id
+           when :order
+             (operating_order.positive? ? operating_order : Float::INFINITY)
+           when :par_price
+             corporation.par_price&.price || 0
+           when :share_price
+             corporation.share_price&.price || 0
+           when :loans
+             corporation.loans.size
+           when :short
+             @game.available_shorts(corporation)
+           when :size
+             corporation.total_shares
+           else
+             @game.player_by_id(@spreadsheet_sort_by)&.num_shares_of(corporation)
+           end, order]
         end
 
+        result.each(&:pop)
```

Comment:
what are you trying to do here

Reply:
We are storing the corp order in 2 ways, but after sorting we've figured out which one we have so we're removing the other.

Reply:
too convoluted